buildscripts/verify-build.sh

```diff
@@ -0,0 +1,228 @@
+    "$MINIO" --config-dir "$MINIO_CONFIG_DIR" server --address=:9003 "http://127.0.0.1:9000${WORK_DIR}/dist-disk1" "http://127.0.0.1:9001${WORK_DIR}/dist-disk2" "http://127.0.0.1:9002${WORK_DIR}/dist-disk3" "http://127.0.0.1:9003${WORK_DIR}/dist-disk4" >"$WORK_DIR/dist-minio-9003.log" 2>&1 &
+
+    sleep 30
+}
+
+function test_make_bucket()
+{
+    bucket_name="mc-test-bucket-$RANDOM"
+    assert_run_mc_command mb "${SERVER_ALIAS}/${bucket_name}"
+    assert_run_mc_command rm "${SERVER_ALIAS}/${bucket_name}"
+}
+
+function setup()
+{
+    assert_run_mc_command mb "${SERVER_ALIAS}/${BUCKET_NAME}"
+}
+
+function teardown()
+{
+    assert_run_mc_command rm --force --recursive "${SERVER_ALIAS}/${BUCKET_NAME}"
+}
+
+function test_put_object()
+{
+    object_name="mc-test-object-$RANDOM"
+    assert_run_mc_command cp "${FILE_1_MB}" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+    assert_run_mc_command rm "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+}
+
+function test_put_object_multipart()
+{
+    object_name="mc-test-object-$RANDOM"
+    assert_run_mc_command cp "${FILE_65_MB}" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+    assert_run_mc_command rm "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+}
+
+function test_get_object()
+{
+    object_name="mc-test-object-$RANDOM"
+    assert_run_mc_command cp "${FILE_1_MB}" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+    assert_run_mc_command cp "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}" "${object_name}.downloaded"
+    assert_run_mc_command rm "${object_name}.downloaded" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+}
+
+function test_get_object_multipart()
+{
+    object_name="mc-test-object-$RANDOM"
+    assert_run_mc_command cp "${FILE_65_MB}" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+    assert_run_mc_command cp "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}" "${object_name}.downloaded"
+    assert_run_mc_command rm "${object_name}.downloaded" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+}
+
+function run_test()
+{
+    case "$1" in
+        "fs") start_minio_fs ;;
+        "xl") start_minio_xl ;;
+        "dist") start_minio_dist ;;
+        *) echo "unknown mode: $1"
+           return 1
+    esac
+
+    test_make_bucket
+
+    setup
+
+    test_put_object
+    test_put_object_multipart
+    test_get_object
+    test_get_object_multipart
+
+    teardown
+}
+
+function main()
+{
+    mkdir "$WORK_DIR"
+    mkdir "$MINIO_CONFIG_DIR"
+    mkdir "$MC_CONFIG_DIR"
+
+    if ! base64 /dev/urandom | head -c 1048576 >"$FILE_1_MB"; then
+        echo "unable to create 1 MiB file"
+        exit 1
+    fi
+
+    if ! base64 /dev/urandom | head -c 68157440 >"$FILE_65_MB"; then
+        echo "unable to create 65 MiB file"
+        exit 1
+    fi
+
+    create_minio_config
+
+    echo "downloading mc"
+    assert_download_mc
+
+    cd "$WORK_DIR"
+
+    echo "running test for FS setup"
+    ( run_test "fs" )
+    rv=$?
+    killall minio
```

Comment:
@balamurugana, can you make run_test returns an array of minio PIDs, it would be better if we kill only Minio servers created by this test script.

Reply:
Done